src/internal/Get-MsGraphResults.ps1

```diff
@@ -433,7 +530,32 @@ function Get-MsGraphResults {
                             $uriQueryEndpointUniqueId.Path = ([IO.Path]::Combine($uriQueryEndpointUniqueId.Path, $id))
                         }
                         if ($DisableUniqueIdDeduplication -or $hashUri.Add($uriQueryEndpointUniqueId.Uri)) {
-                            if (!$DisableGetByIdsBatching -and $id -match '^[{]?[0-9a-fA-F]{8}-([0-9a-fA-F]{4}-){3}[0-9a-fA-F]{12}[}]?$' -and $uriQueryEndpoint.Uri.Segments.Count -eq 3 -and $uriQueryEndpoint.Uri.Segments[2] -in ('directoryObjects', 'users', 'groups', 'devices', 'servicePrincipals', 'applications') -and ($QueryParametersFinal.Count -eq 0 -or ($QueryParametersFinal.Count -eq 1 -and $QueryParametersFinal.ContainsKey('$select')))) {
+                            if ($EnableInFilter -and $id -match '^[{]?[0-9a-fA-F]{8}-([0-9a-fA-F]{4}-){3}[0-9a-fA-F]{12}[}]?$') {
```

Comment:
The long-term goal would be to enable the In Filter method by default right? Any objection to inverting this switch to follow the same pattern as the other batching switches? So something like "$DisableInFilterBatching". If it is still too early to make it the default, then perhaps we put a not in the parameter description that this parameter is experimental and may change?

Reply:
I would propose to keep the enable and include the experimental warning. As soon as we can set that as "not experimental" we can switch to a DisableInFilter behavior. Will follow up with Graph team.